### PR TITLE
Overflow menu component

### PR DESCRIPTION
**Pull request by zoheiry**

### CODE REVIEW

File: src/components/Dropdown/Dropdown.js [hunk 1]
```diff
@@ -41,25 +36,22 @@ const ToggleIcon = styled(Icon).attrs({
 `;
 
 const DropdownPanel = styled.div`
-  position: absolute;
-  display: ${(props) => (props.open ? 'block' : 'none')};
+  display: block;
```
Comment (frontalicious-martijn) [inline src/components/Dropdown/Dropdown.js]: No longer needed I think. Div is a block by default right?

File: src/components/OverflowMenu/OverflowMenu.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      menuOpen: props.openByDefault
+    };
+  }
+
+  getColorFromAppearance = () => {
+    const { appearance, theme } = this.props;
+    switch (appearance) {
```
Comment (frontalicious-martijn) [inline src/components/OverflowMenu/OverflowMenu.js]: Might be nicer to do this:
```
const colors = {
  primary: theme.skyBlue,
  secondary: theme.stoneGrey,
  ...
}

getColorFromAppearance = () =>
  colors[this.props.appearance] || this.props.theme.skyBlue;
'''

File: src/components/ToggleMenu/ToggleMenu.js [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    this.props.onClose();
+    this.setState({ open: false }, () => {
+      if (this.props.closeOnOutsideClick) {
+        document.removeEventListener('click', this.handleClose);
+        document.removeEventListener('keydown', this.handleKeyDown);
+      }
+    });
+  }
+
+  toggleMenu = (event) => {
```
Comment (frontalicious-martijn) [inline src/components/ToggleMenu/ToggleMenu.js]: Could be simplified:
```
toggleMenu = () => 
    this.state.open ? this.handleClose(event) : this.handleOpen(event);
```

File: src/components/ToggleMenu/ToggleMenu.js [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  onClose: PropTypes.func,
+  closeOnOutsideClick: PropTypes.bool,
+  menuOffset: PropTypes.string, // distance between the menu and the trigger
+};
+
+ToggleMenu.defaultProps = {
+  openByDefault: false,
+  position: 'bottom',
+  preventOutOfBounds: false,
+  onOpen: () => {},
```
Comment (frontalicious-martijn) [inline src/components/ToggleMenu/ToggleMenu.js]: I prefer to use this: `opOpen: () => null`, so the (my) code formatter does not put the `}` on the next line.

File: src/components/OverflowMenu/OverflowMenu.js (hunk 2 above)
Comment (zoheiry) [reply src/components/OverflowMenu/OverflowMenu.js]: That makes total sense!

I also think we should rename the color variables to match the appearance

**Pull request merged**